test/box-luatest/gh_7448_net_box_cancelled_fiber_error_test.lua

```diff
@@ -0,0 +1,31 @@
+local server = require('test.luatest_helpers.server')
+local fiber = require('fiber')
+local net = require('net.box')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{alias = 'default'}
+    g.server:start()
+    g.conn = net.connect(g.server.net_box_uri)
+end
+
+g.after_all = function()
+    g.conn:close()
+    g.server:drop()
+end
+
+g.test_conn_cancelled_fiber_error = function()
+    t.assert(g.conn:ping())
+    t.assert(g.conn._fiber:is_system())
+
+    g.conn._fiber:cancel()
+    -- Fiber kill is performed asynchronously. Half of the
+    -- second is usually enouth to kill it. But system fiber
+    -- is not supposed to be killed anyway.
+    fiber.sleep(1)
```

Comment:
There will be no need for that `fiber.sleep()` as soon as all needed fibers will be marked as system ones. I'm going to walk through `fiber.info` and try to kill system fibers by their names, after which check if they are not `dead`.